libvmi/msr-index.h

```diff
@@ -0,0 +1,36 @@
+/* The LibVMI Library is an introspection library that simplifies access to
+ * memory in a target virtual machine or in a file containing a dump of
+ * a system's physical memory.  LibVMI is based on the XenAccess Library.
+ *
+ * Author: Tamas K Lengyel (tamas.lengyel@zentific.com)
+ *
+ * This file is part of LibVMI.
+ *
+ * LibVMI is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * LibVMI is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibVMI.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef MSR_INDEX_H
+#define MSR_INDEX_H
+
+#include "libvmi.h"
+
+extern const reg_t msr_all[];
```

Comment:
@tklengyel I also wanted your review about moving `msr-index` out of the Xen driver, to be used by KVM as well.
I had to declare the arrays as `extern` to use them multiple times in the code, and give them an implementation in `libvmi/msr-index.c`